src/Simplex_tree/include/gudhi/Simplex_tree/Simplex_tree_siblings.h

```diff
@@ -78,10 +76,11 @@ class Simplex_tree_siblings {
    * between input filtration_value and the value already 
    * present in the node.
    */
-  void insert(Vertex_handle v, Filtration_value filtration_value) {
+  void insert(Vertex_handle v, const Filtration_value& filtration_value) {
     auto ins = members_.emplace(v, Node(this, filtration_value));
-    if (!ins.second && filtration(ins.first) > filtration_value)
-      ins.first->second.assign_filtration(filtration_value);
+    if (!ins.second){
+      unify_births(ins.first->second.filtration_raw(), filtration_value);
+    }
```

Comment:
Would something like this work and reduce copying a bit?
```c++
template<class Filt>
void insert(Vertex_handle v, Filt&& filtration_value) {
  auto ins = members_.try_emplace(v, this, std::forward<Filt>(filtration_value));
  if (!ins.second)
    unify_births(ins.first->second.filtration_raw(), filtration_value);
}
```
(this reuses `filtration_value` after `forward` only in the case where `try_emplace` did not use it)

Reply:
This method is literally never used somewhere in Gudhi. And as the "Siblings" can theoretically be accessed from outside, but are not really supposed to be, I am not even sure that some outside code uses it. So I was tempted to ask if you don't want to remove it? Otherwise, I will add a test calling it in the unit test.

Reply:
I looked at the places were Nodes are inserted in the members. There are not so many of them where the filtration values are also updated. More or less just in `insert_simplex_raw` and `rec_insert_simplex_and_subfaces_sorted`. Which is probably the reason why this method got forgotten. I can use it there if you want or just remove it. Or add a second version where filtration values are not updated (but perhaps other things instead)?

Reply:
For just 2 uses, I guess it is ok to duplicate the logic.
Although if we can pack emplace+unify_births+update_simplex_tree_after_node_insertion in one function to replace those 2 cases, that could be nice.
As you like.

Reply:
Ok, I can try unify all this. It would make the methods more readable.

Reply:
Hm, the only thing is that the siblings are not really aware of the simplex tree they are contained in. So, calling `update_simplex_tree_after_node_insertion` from the sibling would be difficult. But I could move the insert method for siblings out into the simplex tree instead as a private method.

Reply:
Ok, there are some `if constexpr` in it, but the functions calling the insert are quite simpler now. What do you think?